### ISSUE TITLE
sd command, blackboxes param never used - dead or not-implemented feature?

### ISSUE THREAD

**Issue by anz-gordonj7** [anz-bank/sysl#334]
the params.blackboxes map is only ever set to not-nil in the unit tests (https://github.com/anz-bank/sysl/blob/da04e253b58f676870e5a7ed2ec81c9dcef9ee14/sysl2/sysl/seqs.go#L106-L111)

Is this a missing command line argument? or not completed feature? or something to be removed?

**Comment by springwiz** [comment]
blackboxes are either loaded from the sysl file or the command line and I don't think is up for removal. Please review the test cases related to blackboxes in the sysl_test.go.

**Comment by anz-gordonj7** [comment]
according to https://github.com/anz-bank/sysl/blob/da04e253b58f676870e5a7ed2ec81c9dcef9ee14/sysl2/sysl/codegen.go#L305-L318 there is no command line argument for it, I'm trying to determine why that is.

**Comment by springwiz** [comment]
This is from seqs.go prior to your changes:

returnValues.blackboxesFlag = sd.Flag("blackbox",
		"Input blackboxes in the format App <- Endpoint=Some description, "+
			"repeat '-b App <- Endpoint=Some description' to set multiple blackboxes",
	).Short('b').StringMap()

Not sure why you are not able to see it.

**Comment by anz-gordonj7** [comment]
That is not the value being checked in the first message which I'm asking about. ***cmdContextParam.blackboxes** is only set to not-nil in the unit tests, there is no way it can be set to anything else in normal execution.

**Comment by springwiz** [comment]
Yeah the way the blackboxes are passed from test cases is slightly different from the way it is invoked through cmd line. The test case would unpack those parameters before passing them.
Refer to DoConstructSequenceDiagramsWithParams in seqs_test.go.

**Comment by anz-gordonj7** [comment]
OK, Only 1 test actually uses that, I'd suggest we rchange the tests to go through the same flags as the command line (i.e have the test reconstruct the blackboxFlags map instead of going through different codepaths that the real command would do.

**Comment by ChloePlanet** [comment]
This issue is more of code reorganization.
